# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentUtils.h
# Commit: df6f98df9559
# Full Hash: df6f98df955906841a1d877700a6ffb7a6f8d78c
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-08-03 03:42:48
# Regressor Bug: 1781129
# File Overlap Count: 1
# Description:
#   Bug 1781129 - Part 3: Remove Shmem overload of GetSurfaceData, r=edgar
#   
#   After the previous changes there was only one consumer left of the Shmem
#   version of GetSurfaceData, which could easily be changed to use BigBuffer,
#   removing the need for that overload.
# ==============================================================================

diff -r 905393f66985 -r df6f98df9559 dom/base/nsContentUtils.h
--- a/dom/base/nsContentUtils.h	Tue Aug 02 18:09:40 2022 +0000
+++ b/dom/base/nsContentUtils.h	Tue Aug 02 18:09:41 2022 +0000
@@ -197,8 +197,6 @@
 namespace ipc {
 class BigBuffer;
 class IProtocol;
-class IShmemAllocator;
-class Shmem;
 }  // namespace ipc
 
 namespace gfx {
@@ -2920,14 +2918,6 @@
   static mozilla::Maybe<mozilla::ipc::BigBuffer> GetSurfaceData(
       mozilla::gfx::DataSourceSurface&, size_t* aLength, int32_t* aStride);
 
-  /*
-   * Get the pixel data from the given source surface and fill it in Shmem.
-   * The length and stride will be assigned from the surface.
-   */
-  static mozilla::Maybe<mozilla::ipc::Shmem> GetSurfaceData(
-      mozilla::gfx::DataSourceSurface& aSurface, size_t* aLength,
-      int32_t* aStride, mozilla::ipc::IShmemAllocator* aAlloc);
-
   static mozilla::Maybe<mozilla::dom::IPCImage> SurfaceToIPCImage(
       mozilla::gfx::DataSourceSurface&);
   static already_AddRefed<mozilla::gfx::DataSourceSurface> IPCImageToSurface(